environments/environment_factory.py

```diff
@@ -0,0 +1,59 @@
+"""Environment factory class. Given a valid environment name and its constructor args, returns an instantiation of it
+"""
+from .environments import Environment
+# TODO(Oleguer): Fix this, its not very nice to have to import all classes
+from .pendulum import Pendulum
+from .spring import Spring
+
```

Comment:
os.path.dirname(os.path.dirname( 2 times is an error or intended?

Reply:
It's intended because dirname takes the name of the directory of the file. So if `__file__ = 'Hamiltonian-Generative-Networks/networks/inference_net.py'`, then `os.path.dirname(__file__) = Hamiltonian-Generative-Networks/networks/`, and using it twice gives `os.path.dirname(os.path.dirname(__file__)) = 'Hamiltonian-Generative-Networks'` which is the root of the project.